File: binarysearch/Repeated-K-Length-Substrings/Repeated-K-Length-Substrings.cpp
// https://helloacm.com/count-the-repeated-k-length-substrings/
// https://binarysearch.com/problems/Repeated-K-Length-Substrings
// EASY, STRING, HASH TABLE

int solve(string s, int k) {
    unordered_map<string, int> data;
    string cur = "";
    for (int i = 0; i < s.size(); ++ i) {
        cur += s[i];
        if (cur.size() > k) {
            cur = cur.substr(1);
        }
        data[cur] ++;
    }
    return count_if(begin(data), end(data), [](auto &x) { return x.second > 1;});
}

File: binarysearch/Repeated-K-Length-Substrings/Repeated-K-Length-Substrings.py
# https://helloacm.com/teaching-kids-programming-repeated-k-length-substrings-sliding-window/
# https://binarysearch.com/problems/Repeated-K-Length-Substrings
# EASY, SLIDING WINDOW

class Solution:
    def solve(self, s, k):
        data = defaultdict(int)        
        for i in range(len(s) - k + 1):
            data[s[i:i+k]] += 1
        return len(list(filter(lambda x: x > 1, data.values())))
